packages/x-components/src/x-modules/url/store/actions/update-url.action.ts

```diff
@@ -0,0 +1,102 @@
+import { MappedParam, MappedParams, UrlXStoreModule } from '../types';
+
+/**
+ * Default implementation for the {@link UrlActions.updateUrl}.
+ *
+ * @param context - The {@link https://vuex.vuejs.org/guide/actions.html | context} of the actions,
+ * provided by Vuex.
+ *
+ * @public
+ */
+export const updateUrl: UrlXStoreModule['actions']['updateUrl'] = ({
+  getters: {
+    urlParams: { page, ...params }
+  }
+}) => {
+  const currentUrl = window.location.href;
+  const url = new URL(currentUrl);
+
+  clearUrlParams(url, params);
+  urlSetPage(url, page);
+  urlSetParams(url, params);
+
+  window.history.replaceState({ ...window.history.state }, document.title, url.href);
```

Comment:
Are the rest of the actions (loading the state from the URL, pushing the state instead of replacing it) going to be added in another PR?